internal/backend/runtime/omni/controllers/omni/cluster_machine_config_test.go

```diff
@@ -256,6 +258,136 @@ func (suite *ClusterMachineConfigSuite) TestGenerationError() {
 	)
 }
 
+func (suite *ClusterMachineConfigSuite) TestConfigEncodingStability() {
+	suite.startRuntime()
+
+	suite.Require().NoError(suite.state.Create(suite.ctx, siderolink.NewConnectionParams(resources.DefaultNamespace, siderolink.ConfigID)))
+
+	suite.Require().NoError(suite.runtime.RegisterController(omnictrl.NewClusterController()))
+	suite.Require().NoError(suite.runtime.RegisterController(omnictrl.NewMachineSetController()))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewSchematicConfigurationController(&imageFactoryClientMock{})))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewClusterMachineConfigController(imageFactoryHost, nil, 8090)))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewSecretsController(nil)))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewClusterStatusController(false)))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewTalosUpgradeStatusController()))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewClusterConfigVersionController()))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewMachineConfigGenOptionsController()))
+
+	clusterName := "test-config-encoding-stability"
+	initialVersion := "1.2.1"
+	upgradeVersions := []string{
+		"1.3.0",
```

Comment:
I really like this, but I wonder if we should add one more dimension on `initialVersion`, what I mean is that today we test:

* `1.2.1` -> `1.10.0` 

But we can test also any `1.x` -> `latest`. 

We can use Talos's `pkg/machinery` version to find the "latest".

Reply:
Addressed both, see the sample test output in the PR description.